### PR TITLE
Fix Timedelta floordiv, rfloordiv with offset, fix td64 return types

### PR DESCRIPTION
**Pull request by jbrockmendel**
Fixes xfails introduced in #19752.

### CODE REVIEW

File: doc/source/whatsnew/v0.23.0.txt [hunk 1]
```diff
@@ -716,6 +716,8 @@ Datetimelike
 - Bug in :class:`Timestamp` and :func:`to_datetime` where a string representing a barely out-of-bounds timestamp would be incorrectly rounded down instead of raising ``OutOfBoundsDatetime`` (:issue:`19382`)
 - Bug in :func:`Timestamp.floor` :func:`DatetimeIndex.floor` where time stamps far in the future and past were not rounded correctly (:issue:`19206`)
 - Bug in :func:`to_datetime` where passing an out-of-bounds datetime with ``errors='coerce'`` and ``utc=True`` would raise ``OutOfBoundsDatetime`` instead of parsing to ``NaT`` (:issue:`19612`)
+- Bug in :func:`Timedelta.__add__`, :func:`Timedelta.__sub__` where adding or subtracting a ``np.timedelta64`` object would return another ``np.timedelta64`` instead of a ``Timedelta`` (:issue:`19738`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.23.0.txt]: would not be averse to a PR separating this whatsnew to Datetime, Period, Timedelta sections (so each is not so long)

**Review by jreback — APPROVED** [review]
ex my below comment, lgtm. ping on green.

**Comment by jbrockmendel** [comment]
ping

**Comment by jreback** [comment]
thanks!

**Pull request merged**